pkg/networkservice/common/monitor/server_test.go

```diff
@@ -0,0 +1,268 @@
+package monitor
+
+import (
+	"context"
+	"fmt"
+	"github.com/golang/protobuf/ptypes/empty"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/metadata"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/trace"
+)
+
+type testMonitorServer struct {
+	events       []*networkservice.ConnectionEvent
+	eventChannel chan *networkservice.ConnectionEvent
+	ctx          context.Context
+	cancel       context.CancelFunc
+}
+
+func newTestMonitorServer() *testMonitorServer {
+	ctx, cancel := context.WithCancel(context.Background())
+	return &testMonitorServer{
+		eventChannel: make(chan *networkservice.ConnectionEvent, 10),
+		ctx:          ctx,
+		cancel:       cancel,
+	}
+}
+
+func (t *testMonitorServer) Send(evt *networkservice.ConnectionEvent) error {
+	t.events = append(t.events, evt)
+	t.eventChannel <- evt
+	return nil
+}
+
+func (t *testMonitorServer) SetHeader(metadata.MD) error {
+	return nil
+}
+
+func (t *testMonitorServer) SendHeader(metadata.MD) error {
+	return nil
+}
+
+func (t *testMonitorServer) SetTrailer(metadata.MD) {
+}
+
+func (t *testMonitorServer) Context() context.Context {
+	return t.ctx
+}
+
+func (t *testMonitorServer) SendMsg(m interface{}) error {
+	return nil
+}
+
+func (t *testMonitorServer) RecvMsg(m interface{}) error {
+	return nil
+}
+
+func (t *testMonitorServer) BeginMonitoring(server networkservice.MonitorConnectionServer, segmentName string) {
+	go func() {
+		_ = server.MonitorConnections(
+			&networkservice.MonitorScopeSelector{
+				PathSegments: []*networkservice.PathSegment{{Name: segmentName}},
+			}, t)
+	}()
+}
+
+func (t *testMonitorServer) WaitEvents(ctx context.Context, count int) {
+	for {
+		if len(t.events) == count {
+			logrus.Infof("Waiting for events %v, but has %v", count, len(t.events))
```

Comment:
Info line is strange, because `count` and `len(t.events)` are equal if it prints